src/System.AppContext/src/System/AppContext.cs

```diff
@@ -80,7 +77,7 @@ public static void SetSwitch(string switchName, bool isEnabled)
             if (switchName.Length == 0)
                 throw new ArgumentException(SR.Argument_EmptyName, "switchName");
 
-            lock (s_syncLock)
+            lock (s_switchMap)
```

Comment:
Nit: As long as you're moving work out of the body of a lock, you could do the same for this one (determining which value to store), though it's unlikely to make any measurable impact.